src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -184,32 +181,41 @@ private void addRolesToUser(User user, Connection connection) {
         }
     }
 
-    private Long getRoleIdByName(Role.RoleName roleName, Connection connection) {
+    private void getRoleIdByName(Role role, Connection connection) {
         String query = "SELECT role_id FROM roles WHERE role_name = ?";
+        Role.RoleName roleName = role.getRoleName();
         try (PreparedStatement statement = connection.prepareStatement(query);) {
             statement.setString(1, roleName.name());
             ResultSet resultSet = statement.executeQuery();
-            while (resultSet.next()) {
-                return resultSet.getLong("role_id");
+            if (resultSet.next()) {
+                role.setId(resultSet.getLong("role_id"));
```

Comment:
I think that logically it is more correct to have the following method signature:
`private Long getRoleIdByName(Role.RoleName roleName, Connection connection)`

let's return id if we have an item in the resultSet, otherwise throw an exception